deployment/centos-package-x64/Dockerfile

```diff
@@ -1,4 +1,4 @@
-FROM centos:7
+FROM centos:8
```

Comment:
Could one still want to build on CentOS 7?

I'm honestly not entirely sure what this Dockerfile is meant to do.  Does it simpley provide a means of running JF inside of a CentOS 8 docker container for those that want that kind of deployment?

How many of these exist?  Meaning docker images of JF for different O/Ses?  Don't they all end up supplying the same functionality: JF in a docker container.

Reply:
1. This should be able to still build on CentOS 7, since I removed the package `yum-plugins-core`, this package changed in CentOS 8 since the move from `dnf` to `yum`.  I still need to validate the build, which I can do if that is necessary.

2. From my understanding this Dockerfile and others within the `deployment` directory are all for package builds.  This one being for the RPM build for CentOS, I believe @joshuaboniface is working restructuring the entire build process now.  I was just trying to get a working build for CentOS 8 based on the issue #2563 

3. I am assuming all of the ones you see in the `deployment` directory are for building the package for their respective distribution.  I am not sure the reason for it, but I believe there was one.

Reply:
I see.  I suppose a zillion Dockerfiles is one way to do it.  Typically [mock](https://github.com/rpm-software-management/mock) is the defacto/canonical way of building RPMs in a "clean-room" environment.  It more-or-less achieves the same result without the need to continue maintaining Dockerfiles.

So given that, shouldn't a CentOS 8 building Dockerfile be in addition to the CentOS 7 one, not replacing it?  There are still a **lot** of people on CentOS 7.  Maybe more on CentOS 7 than have moved to 8 even still.  I certainly don't mind seeing this project having sooner-than-later support for 8, to be sure.  It just shouldn't be at the expense of continuing to support 7.  IMHO.

Reply:
I agree we probably should continue to support CentOS 7.  My original intent was to get the builds working on CentOS 8, which I was able to, but I was unsure which version JF wanted to use for their builds.

The factor that was causing the issue from build on CentOS 8 was primarily the package `yum-plugins-core`.  I removed this package because it seemed the build never used any of the plugins offered by this package.

If the team is wanting to support both versions I, personally, don't know the best way to handle that unless it's using Compose or just a separate Dockerfile.

I don't know much about Mock, but just from the README it seems pretty nice for maintaining builds for RPMs.